### PR TITLE
ref(radio-group): Add disabledChoices prop

### PR DESCRIPTION
**Pull request by priscilawebdev**
- [x] Add a new optional prop called `disabledChoices`
- [x] Convert Test from enzyme to React Testing Library
- [x] Add a new test to test one single disabled choice

Preview:

image

Requires https://github.com/getsentry/getsentry/pull/7061

### CODE REVIEW

File: static/app/components/forms/controls/radioGroup.tsx [hunk 1]
```diff
@@ -46,32 +53,37 @@ const RadioGroup = <C extends string>({
...
-            <Description>{description}</Description>
-          </React.Fragment>
-        )}
-      </RadioLineItem>
-    ))}
+    {choices.map(([id, name, description], index) => {
+      const disabledChoice = disabledChoices.find(([choiceId]) => choiceId === id);
+      const disabled = !!disabledChoice || groupDisabled;
+      return (
+        <Tooltip key={index} disabled={!disabledChoice?.[1]} title={disabledChoice?.[1]}>
```
Comment (markstory) [inline static/app/components/forms/controls/radioGroup.tsx:60]: Isn't this a lot of tooltip components that will rarely be used? Should we only emit these components when they will be active?

File: tests/js/spec/components/deprecatedforms/radioGroup.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import {mountWithTheme, screen, userEvent} from 'sentry-test/reactTestingLibrary';
+
+import RadioGroup from 'sentry/components/forms/controls/radioGroup';
```
Comment (markstory) [inline tests/js/spec/components/deprecatedforms/radioGroup.spec.tsx:3]: Should this test module be in `deprecatedforms`?

Comment (priscilawebdev) [reply tests/js/spec/components/deprecatedforms/radioGroup.spec.tsx:3]: oh no... I think I missed this when rebasing

File: static/app/components/forms/controls/radioGroup.tsx [hunk 3]
```diff
@@ -46,32 +53,51 @@ const RadioGroup = <C extends string>({
...
+          </RadioLineItem>
+        </React.Fragment>
+      );
+
+      if (!!disabledChoice?.[1]) {
+        return (
+          <Tooltip
+            key={index}
+            disabled={!disabledChoice?.[1]}
+            title={disabledChoice?.[1]}
```
Comment (matejminar) [inline static/app/components/forms/controls/radioGroup.tsx:92]: nit: Can we extract these 3 to a variable that's more descriptive? Perhaps disabledReason?

**Review by matejminar — APPROVED** [review]
Have we considered extending the `choices` prop with disabled/reason? Although it might be clunky as it is an array and not an object 🤔

**Comment by priscilawebdev** [comment]
@matejminar yes in the beginning I have done that, but since it's not an object, I thought that having a separate prop would be better...We can maybe refactor this component later and update the instances using it

**Pull request merged**